nengo_viz/swi.py

```diff
@@ -169,7 +169,7 @@ def swi_favicon_ico(self):
 
     def do_POST(self):
         self.clear_headers()
-        data = self.rfile.read(int(self.headers['Content-Length']))
+        data = self.rfile.read(int(self.headers['Content-Length'])).decode('ascii')
```

Comment:
Possibly not, but the whole rest of the code assumes it. 

(and it might even be correct to assume, because that's parsing the POST request, which has some rules about its format.  But I think it's a separate issue to actually check all the unicode issues.)